### PR TITLE
Improve variant spec coverage

### PR DESCRIPTION
**Pull request by mamhoff**
## Summary

This adds some specs for a few instance methods on Spree::Variant. No functional changes.

## Checklist

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [x] I have added automated tests to cover my changes.
- [ ] ~I have attached screenshots to demo visual changes.~
- [ ] ~I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~
- [ ] ~I have updated the readme to account for my changes.~

### CODE REVIEW

**Review by tvdeyen — APPROVED** [review]
Thanks

**Review by kennyadsl — APPROVED** [review]
Danke Martin!

File: core/spec/models/spree/variant_spec.rb [hunk 1]
```diff
@@ -45,6 +45,28 @@
       variant_with_same_sku = build(:variant, sku: variant.sku)
       expect(variant_with_same_sku).to be_invalid
     end
+
+    context "should have a price if master variant" do
```
Comment (kennyadsl) [inline core/spec/models/spree/variant_spec.rb:49]: Just one nit re-reviewing: this sounds a bit off as a context title, isn't it?

Comment (mamhoff) [reply core/spec/models/spree/variant_spec.rb]: Fixed!

**Review by jarednorman — APPROVED** [review]

File: core/app/models/spree/variant.rb [hunk 2]
```diff
@@ -211,7 +211,7 @@ def deleted?
     # Assign given options hash to option values.
     #
     # @param options [Array] array of hashes with a name and value.
```
Comment (waiting-for-dev) [inline core/app/models/spree/variant.rb:213]: The doc is also wrong, as it should be `[Hash]`. Nice if we fix it now.

**Review by waiting-for-dev — APPROVED** [review]
Minor suggestion to go the extra mile 🙂

File: core/app/models/spree/variant.rb (hunk 2 above)
Comment (mamhoff) [reply core/app/models/spree/variant.rb:213]: I think the brackets only indicate that this is the type part of the line, not that it's an array. I can be specific and do `[Array<Hash>]`.

Comment (waiting-for-dev) [reply core/app/models/spree/variant.rb]: Oh, yeah, sorry, I was confused. Thanks for your change.

**Pull request merged**